ddd-koans/Value Object/Static Factory & Reflect Domain Representation/src/Main.kt

```diff
@@ -46,4 +23,16 @@ class ValidTimeSignature(private val numerator: Int, private val denominator: In
     }
 
     override fun toString(): String = "TimeSignature($numerator/$denominator)"
-}
+
+    companion object {
+        fun of(timeSignature: String): TimeSignature {
```

Comment:
I tried below, regex-less implementation and it fails:

```
     timeSignature
                .trim()
                .takeIf { it.length == 3 }
                ?.takeIf { it[0].isDigit() && it[2].isDigit() }
                ?.let { TimeSignature(it[0].toInt(), it[2].toInt()) }
                ?: throw RuntimeException()
```

It fails with message `Numerator must be integer between 1 and 32` which comes from the `TimeSignature` constructor so it means that some exception is thrown, so why didn't it pass the test? I think it's for `0/0` case

I tried this approach as not always regex is completely understandable by the audience 😉

Reply:
Why have you assumed that length == 3  is the proper approach? If it is 3/32 it is 4 already?

Do you want to get rid of regex completely? It is just in our solution. There is nor an advice or a recommendation about using it.

Reply:
if it's not a recommendation then I'm ok

Reply:
and I made a wrong length assumption, just tried the simplest, naive approach

Reply:
This is how it looks now: "TimeSignature created using "1/1" should be created, but exception was thrown"
It is hard to point exact error in the message as in that code it is tricky 😸